viseron/components/yolo/object_detector.py

```
@@ -0,0 +1,122 @@
+
+    def postprocess(self, results):
+        """Return yolo detections as DetectedObject."""
+
+        objects = []
+
+        for result in results:
+            classes_names = result.names
+
+            for box in result.boxes:
+                cls = int(box.cls[0])
+                [x1, y1, x2, y2] = box.xyxy[0]
+                x1, y1, x2, y2 = int(x1), int(y1), int(x2), int(y2)
+                objects.append(
+                    DetectedObject.from_absolute(
+                        label=classes_names[cls],
+                        confidence=float(box.conf),
+                        x1=x1,
+                        y1=y1,
+                        x2=x2,
+                        y2=y2,
+                        frame_res=self._camera.resolution,
+                        model_res=result.orig_shape[::-1],
+                    )
+                )
+        return objects
+
+    def return_objects(self, frame):
+        """Perform object detection."""
+        if self._detector is None:
+            return []
+        try:
+            results = self._detector.predict(
+                frame,
+                conf=self._config[CONFIG_MIN_CONFIDENCE],
+                iou=self._config[CONFIG_IOU],
+                half=self._config[CONFIG_HALF_PRECISION],
+                device=self._config[CONFIG_DEVICE],
+                verbose=False,
+            )
+        except ValueError as exception:
+            LOGGER.error(f"Error calling yolo prediction: {exception}")
+            LOGGER.error(
+                "Inference has been disabled for this camera. "
+                "Check yolo configuration in config.yaml"
+            )
+            self._detector = None
+            return []
+        except Exception as exception:
+            LOGGER.error(f"Error during yolo prediction: {exception}")
+            LOGGER.error(
+                "Inference has been disabled for this camera. "
+            )
+            self._detector = None
+            return []
```

I feel like this could get confusing, if a temporary error occurs and the user is not looking at the logs at that particular moment, object detection will essentially silently stop.

I think its better to just let the exception be thrown, Viseron will automatically restart the object detector thread after a delay